tests/integration/MyObservations.test.js

```diff
@@ -0,0 +1,55 @@
+// These test ensure that My Observation integrates with other systems like
+// remote data retrieval and local data persistence
+
+import React from "react";
+import factory, { makeResponse } from "../factory";
+import { render, waitFor, within, act } from "@testing-library/react-native";
+import { NavigationContainer } from "@react-navigation/native";
+import AccessibilityEngine from "react-native-accessibility-engine";
+import ObsList from "../../src/components/Observations/ObsList";
+import ObservationProvider from "../../src/providers/ObservationProvider";
+
+// Mock inaturalistjs so we can make some fake responses
+jest.mock( "inaturalistjs" );
+import inatjs from "inaturalistjs";
+
+const renderObsList = async ( ) => waitFor(
+  ( ) => render(
+    <NavigationContainer>
+      <ObservationProvider>
+        <ObsList />
+      </ObservationProvider>
+    </NavigationContainer>
+  )
+);
+
+test( "renders the number of comments from remote response", async ( ) => {
```

Comment:
I'm curious why you chose to render from the remote response instead of the local, in-memory database. Since we're only rendering the list of observations once they're stored in Realm, wouldn't rendering local responses be a more realistic integration test?

Reply:
Hm, maybe the test description is misleading. What I meant was "render the number of comments from a remote response that gets persisted locally and made available to the ObsList component." That's what's happening, but that seemed excessively wordy.

Again, it's somewhat subjective, but I interpret "integration" to mean integration between the different parts of the code in your app. If we mock the Realm response here, we ignore the code we have for mapping API responses to Realm objects, so we're not integrating as much of our own code as we could and thus not testing something closer to the actual user experience. With this approach, we're only mocking an input from outside our app (the inatjs response), which then gets run through our mapping code to create Realm records, which are then made available to the component through our own context and providers, so this is testing the integration of all of those things that we wrote. It's also integrating some things that we didn't write, like the internals of Realm, but as long as they work quickly and predictively, I think that's actually a good thing.